lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -491,12 +491,17 @@ def _merge_tracks(self, starting_track, starting_node, ending_track, ending_node
         if starting_track not in self._src or ending_track not in self._src:
             raise RuntimeError("Both tracks need to be part of this group to be merged")
 
-        starting_node = int(starting_node) + 1
+        starting_node = int(starting_node)
         ending_node = int(ending_node)
 
+        assert (
+            starting_track.time_idx[starting_node] < ending_track.time_idx[ending_node]
+        ), "The starting node must occur before the ending node."
```

Comment:
It's good that the assumption was made explicit, but that means I still have to manually swap the starting and ending node. Why can't you just swap it here, given that you already check for the condition?

Reply:
yeah, not sure why I didn't do it that way. probably because the original work was for the widget so my head jumped straight to handling it there for some reason. I'll move the functionality to here